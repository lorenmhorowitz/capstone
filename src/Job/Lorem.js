export const lorem =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer mattis tellus quis turpis gravida sollicitudin. Quisque vehicula lorem consectetur vehicula imperdiet. Nullam non suscipit ex, et dictum tortor. Vivamus eget volutpat elit. Fusce posuere finibus laoreet. Donec tempus orci nec elit condimentum rhoncus. Etiam diam nunc, rutrum sed condimentum nec, gravida ut metus. Nullam tempus aliquet ante sit amet facilisis. Ut et finibus massa. Pellentesque eget elit a neque venenatis scelerisque. Maecenas nec nunc massa. Phasellus eget velit vitae orci gravida venenatis in quis nisl. Phasellus non lectus eu velit tristique suscipit quis nec enim. Quisque venenatis ultricies tortor maximus maximus. Aenean ut pharetra nisl. Quisque porttitor, lorem et euismod congue, felis turpis hendrerit sem, quis tempor purus nisl accumsan dolor. Suspendisse fringilla ex vitae nisl convallis ornare. Quisque porta tempus sollicitudin. Quisque suscipit velit ut elit hendrerit, in varius mauris malesuada. Aenean vitae elementum libero. Sed dignissim porttitor nisl, viverra ornare erat porttitor sit amet. In tincidunt orci nibh, ut fermentum urna volutpat non. Ut condimentum accumsan neque ut facilisis. Nunccongue luctus est a lacinia. Cras a ultricies erat, et tincidunt elit. Suspendisse eget mi vel leo luctus dapibus. Praesent convallis arcu vitae dolor sagittis, porta euismod quam aliquam. Sed vitae sollicitudin eros.";
