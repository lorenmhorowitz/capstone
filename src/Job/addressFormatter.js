export const addressFormatter = (
  location_city,
  location_region,
  location_postal_code
) => {
  let res = "";
  if (location_city !== null && location_city !== "") {
    res += location_city;
  }
  if (location_region !== null && location_region !== "") {
    if (res !== "") res += `, `;
    res += location_region;
  }
  if (location_postal_code !== null && location_postal_code !== "") {
    if (res !== "") res += " ";
    res += location_postal_code;
  }
  return res;
};
