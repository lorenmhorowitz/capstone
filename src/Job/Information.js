import { addressFormatter } from "./addressFormatter";
import CardSlide from "../components/CardSlide";
import Divider from "@material-ui/core/Divider";
import GoogleMap from "../components/GoogleMap";
import Grid from "@material-ui/core/Grid";
import React, { Component, Fragment } from "react";
import Typography from "@material-ui/core/Typography";

import "../css/job.css";

class Information extends Component {
  render() {
    const { jobDetails } = this.props;
    const images = [
      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1"
    ];

    const formattedAddress = addressFormatter(
      jobDetails.location_city,
      jobDetails.location_region,
      jobDetails.location_postal_code
    );
    const googleMapsAddress =
      formattedAddress !== ""
        ? `${jobDetails.location_line_1} ${formattedAddress}`
        : "Vancouver, WA";

    return (
      <Fragment>
        <Grid container style={{ paddingTop: "2em" }}>
          <Grid item xs={4}>
            <div id="grid1">
              <Typography variant="subtitle1">Address:</Typography>
              <Divider style={{ color: "orange" }} />
              <Typography variant="h6">{jobDetails.location_line_1}</Typography>
              <Typography variant="h6">{formattedAddress}</Typography> <p />
              <Typography variant="subtitle1">Client:</Typography>
              <Divider />
              <Typography variant="h6">{jobDetails.name}</Typography> <p />
              <Typography variant="subtitle1">Contact Information: </Typography>
              <Divider />
              <Typography variant="h6">(360) 123-4567</Typography>
              <Typography variant="h6">test@wsu.edu</Typography>
            </div>
          </Grid>
          <Grid item xs={8}>
            <Grid container>
              <Grid item>
                <div id="picture">
                  <CardSlide images={images} width={"100%"} height={"100%"} />
                </div>
              </Grid>
              <Grid item>
                <div id="map">
                  <GoogleMap
                    location={googleMapsAddress}
                    height="100%"
                    width="100%"
                  />
                </div>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Fragment>
    );
  }
}

export default Information;
