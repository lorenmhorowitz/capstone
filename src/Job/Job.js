import React, { Component, Fragment } from "react";
import AppBar from "../components/AppBar";
import Divider from "@material-ui/core/Divider";
import Information from "./Information";
import Loading from "../components/Loading";
import request from "request";
import SideBar from "../components/SideBar";
import Typography from "@material-ui/core/Typography";

import "../css/job.css";

const JOBURL = "https://us-central1-hdqc-capstone.cloudfunctions.net/getJob";

class Job extends Component {
  constructor(props) {
    super(props);
    this.state = {
      jobDetails: {},
      loading: true
    };

    request.post(
      {
        url: JOBURL,
        headers: {
          "Content-Type": "application/json"
        },
        json: true,
        body: {
          key: "username",
          kind: "jobs",
          id: window.location.pathname.replace("/job/", "")
        }
      },
      (error, response, body) => {
        this.setState({
          jobDetails: { ...body },
          loading: false
        });
      }
    );
  }

  render() {
    const loaded = (
      <div id="mainWindow">
        <Typography id="header">Job Information</Typography>
        <Divider id="bar1" />
        <div>
          <Information jobDetails={this.state.jobDetails} />
          <p />
        </div>

        {this.state.jobDetails.hasOwnProperty("active_projects") &&
        this.state.jobDetails.active_projects.roofing ? (
          <Fragment>
            <Typography id="header2">Roofing Information</Typography>
            <Divider id="bar1" />
            <div>
              <p />
            </div>
          </Fragment>
        ) : null}

        {this.state.jobDetails.hasOwnProperty("active_projects") &&
        this.state.jobDetails.active_projects.siding ? (
          <Fragment>
            <Typography id="header">Siding Information</Typography>
            <Divider id="bar1" />
            <div>
              <p />
            </div>
          </Fragment>
        ) : null}

        {this.state.jobDetails.hasOwnProperty("active_projects") &&
        this.state.jobDetails.active_projects.windows ? (
          <Fragment>
            <Typography id="header">Windows Information</Typography>
            <Divider id="bar1" />
            <div />
          </Fragment>
        ) : null}
      </div>
    );

    return (
      <div>
        <AppBar />
        {!this.state.loading &&
        this.state.jobDetails.hasOwnProperty("location_line_1")
          ? loaded
          : null}
        {!this.state.loading &&
        !this.state.jobDetails.hasOwnProperty("location_line_1") ? (
          <div id="mainWindow">
            <Typography id="title1">
              Unable to load job {window.location.pathname.replace("/job/", "")}
              .
            </Typography>
          </div>
        ) : null}

        {this.state.loading ? <Loading /> : null}
        <SideBar />
      </div>
    );
  }
}

export default Job;
