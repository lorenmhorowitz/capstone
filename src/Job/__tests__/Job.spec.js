import React from "react";
import Job from "../Job";
import { shallow } from "enzyme";

describe.skip("Job Component", () => {
  it("renders without crashing", () => {
    let wrapper = shallow(<Job />);
    expect(wrapper).toMatchSnapshot();
  });
});
