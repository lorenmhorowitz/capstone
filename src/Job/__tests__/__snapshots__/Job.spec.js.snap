// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Job />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <WithStyles(AppBar) />
        </div>,
        <WithStyles(Grid)
          container={true}
          id="grid"
        >
          <WithStyles(Grid)
            item={true}
          >
            <div
              id="font1"
            >
              2910 NE 168th Ave
            </div>
            <div
              id="font1"
            >
              Vancouver, WA 98682
            </div>
            <div
              id="font2"
            >
              Arnold Castro
            </div>
            <div
              id="font2"
            >
              (360) 123-4567
            </div>
            <div
              id="font2"
            >
              test@wsu.edu
            </div>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            id="grid2"
            item={true}
          >
            <div
              className="picture"
            >
              <CardSlide
                height={450}
                images={
                  Array [
                    "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                    "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                  ]
                }
                width={750}
              />
            </div>
          </WithStyles(Grid)>
        </WithStyles(Grid)>,
        <WithStyles(CircularIndeterminate) />,
        <WithStyles(Sidebar) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(AppBar) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
            >
              <div
                id="font1"
              >
                2910 NE 168th Ave
              </div>
              <div
                id="font1"
              >
                Vancouver, WA 98682
              </div>
              <div
                id="font2"
              >
                Arnold Castro
              </div>
              <div
                id="font2"
              >
                (360) 123-4567
              </div>
              <div
                id="font2"
              >
                test@wsu.edu
              </div>
            </WithStyles(Grid)>,
            <WithStyles(Grid)
              id="grid2"
              item={true}
            >
              <div
                className="picture"
              >
                <CardSlide
                  height={450}
                  images={
                    Array [
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                    ]
                  }
                  width={750}
                />
              </div>
            </WithStyles(Grid)>,
          ],
          "container": true,
          "id": "grid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <div
                  id="font1"
                >
                  2910 NE 168th Ave
                </div>,
                <div
                  id="font1"
                >
                  Vancouver, WA 98682
                </div>,
                <div
                  id="font2"
                >
                  Arnold Castro
                </div>,
                <div
                  id="font2"
                >
                  (360) 123-4567
                </div>,
                <div
                  id="font2"
                >
                  test@wsu.edu
                </div>,
              ],
              "item": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2910 NE 168th Ave",
                  "id": "font1",
                },
                "ref": null,
                "rendered": "2910 NE 168th Ave",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vancouver, WA 98682",
                  "id": "font1",
                },
                "ref": null,
                "rendered": "Vancouver, WA 98682",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Arnold Castro",
                  "id": "font2",
                },
                "ref": null,
                "rendered": "Arnold Castro",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "(360) 123-4567",
                  "id": "font2",
                },
                "ref": null,
                "rendered": "(360) 123-4567",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test@wsu.edu",
                  "id": "font2",
                },
                "ref": null,
                "rendered": "test@wsu.edu",
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="picture"
              >
                <CardSlide
                  height={450}
                  images={
                    Array [
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                    ]
                  }
                  width={750}
                />
              </div>,
              "id": "grid2",
              "item": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CardSlide
                  height={450}
                  images={
                    Array [
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                    ]
                  }
                  width={750}
                />,
                "className": "picture",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "height": 450,
                  "images": Array [
                    "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                    "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                  ],
                  "width": 750,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <WithStyles(AppBar) />
          </div>,
          <WithStyles(Grid)
            container={true}
            id="grid"
          >
            <WithStyles(Grid)
              item={true}
            >
              <div
                id="font1"
              >
                2910 NE 168th Ave
              </div>
              <div
                id="font1"
              >
                Vancouver, WA 98682
              </div>
              <div
                id="font2"
              >
                Arnold Castro
              </div>
              <div
                id="font2"
              >
                (360) 123-4567
              </div>
              <div
                id="font2"
              >
                test@wsu.edu
              </div>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              id="grid2"
              item={true}
            >
              <div
                className="picture"
              >
                <CardSlide
                  height={450}
                  images={
                    Array [
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                    ]
                  }
                  width={750}
                />
              </div>
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
          <WithStyles(CircularIndeterminate) />,
          <WithStyles(Sidebar) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(AppBar) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
              >
                <div
                  id="font1"
                >
                  2910 NE 168th Ave
                </div>
                <div
                  id="font1"
                >
                  Vancouver, WA 98682
                </div>
                <div
                  id="font2"
                >
                  Arnold Castro
                </div>
                <div
                  id="font2"
                >
                  (360) 123-4567
                </div>
                <div
                  id="font2"
                >
                  test@wsu.edu
                </div>
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                id="grid2"
                item={true}
              >
                <div
                  className="picture"
                >
                  <CardSlide
                    height={450}
                    images={
                      Array [
                        "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                        "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                      ]
                    }
                    width={750}
                  />
                </div>
              </WithStyles(Grid)>,
            ],
            "container": true,
            "id": "grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <div
                    id="font1"
                  >
                    2910 NE 168th Ave
                  </div>,
                  <div
                    id="font1"
                  >
                    Vancouver, WA 98682
                  </div>,
                  <div
                    id="font2"
                  >
                    Arnold Castro
                  </div>,
                  <div
                    id="font2"
                  >
                    (360) 123-4567
                  </div>,
                  <div
                    id="font2"
                  >
                    test@wsu.edu
                  </div>,
                ],
                "item": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2910 NE 168th Ave",
                    "id": "font1",
                  },
                  "ref": null,
                  "rendered": "2910 NE 168th Ave",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vancouver, WA 98682",
                    "id": "font1",
                  },
                  "ref": null,
                  "rendered": "Vancouver, WA 98682",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Arnold Castro",
                    "id": "font2",
                  },
                  "ref": null,
                  "rendered": "Arnold Castro",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "(360) 123-4567",
                    "id": "font2",
                  },
                  "ref": null,
                  "rendered": "(360) 123-4567",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test@wsu.edu",
                    "id": "font2",
                  },
                  "ref": null,
                  "rendered": "test@wsu.edu",
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="picture"
                >
                  <CardSlide
                    height={450}
                    images={
                      Array [
                        "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                        "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                      ]
                    }
                    width={750}
                  />
                </div>,
                "id": "grid2",
                "item": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <CardSlide
                    height={450}
                    images={
                      Array [
                        "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                        "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                      ]
                    }
                    width={750}
                  />,
                  "className": "picture",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "height": 450,
                    "images": Array [
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc11.jpg?w=1000&h=563&crop=1",
                      "https://pmcvariety.files.wordpress.com/2018/07/bradybunchhouse_sc_fi.jpg?w=700&h=393&crop=1",
                    ],
                    "width": 750,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
